cpp/include/cudf/datetime.hpp

```diff
@@ -183,6 +155,41 @@ std::unique_ptr<cudf::column> day_of_year(
   cudf::column_view const& column,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
 
+/**
+ * @brief  Adds or subtracts a number of months from the date time type and returns a
+ * timestamp column that is of the same type as `timestamp_column`.
+ *
+ * For a given row, if the `timestamp_column` or the `months_column` value is null,
+ * the output for that row is null.
+ *
+ * @code{.pseudo}
+ * Example:
+ * timestamp_column = [5/31/20 08:00:00, 5/31/20 00:00:00, 5/31/20 13:00:00, 5/31/20 23:00:00,
+ *                     6/30/20 00:00:01, 6/30/20 14:12:13]
+ * months_columns   = [1               , -1              , -3              , -15             ,
+ *                     -1              , 1]
+ * r = add_months(timestamp_column, months_column)
+ * r[0] is [6/30/20 08:00:00]
+ * r[1] is [4/30/20 00:00:00]
+ * r[2] is [2/29/20 13:00:00]
+ * r[3] is [2/28/19 23:00:00]
+ * r[4] is [5/30/20 00:00:01]
+ * r[5] is [7/30/20 14:12:13]
+ * @endcode
+
+ * @param timestamp_column cudf::column_view of the input datetime values
+ * @param months_column cudf::column_view of the number of months to add or subtract
+ *
+ * @returns cudf::column of datatype `timestamp_column` containing the number of months
+ * before or after the input datetime value.
```

Comment:
I don't think `timestamp_column` is the name of any of our types.

```suggestion
 * @returns cudf::column of timestamp type containing the computed timestamps.
```

Reply:
done